src/registry.rs

```diff
@@ -48,17 +48,35 @@ impl RegistryCore {
 
             self.dim_hashes_by_name.insert(desc.fq_name.clone(), desc.dim_hash);
 
-            desc.id
-        };
+            // If it is not a duplicate desc in this collector, add it to
+            // the collector_id.
+            if let None = id_set.iter().find(|id| **id == desc.id) {
+                id_set.push(desc.id);
+                collector_id += desc.id;
```

Comment:
Is it possible that 2 collectors contain desc ids [1, 4] and [2, 3] respectively, then their `collector_id` become equal?